### ISSUE TITLE
flag --continue=false or default value is not stopping the pipeline (new issue in ^1.5.5)

### ISSUE THREAD

**Issue by AddictArts** [vercel/turbo#2215]
### What version of Turborepo are you using?

1.5.6

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Windows

### Describe the Bug

Turbo cli used to stop execution when a task had an error, now it doesn't in recent versions. Note, forcing `--continue=false` on the command line further doesn't stop turbo cli from going on.

```shell
//:outdated: ERROR: command finished with error: command (\Software\oranges) npm run outdated exited (1)
```

### Expected Behavior

The pipelines stop and the turbo process exits.

### To Reproduce

Add a lint pipeline and have another depend on it. Put a lint error in the code. Run the pipeline. Or use something like `npm outdated` and use an older package. Make this a dependency of your current pipeline. You'll see the outdated packages, but turbo will continue to execute.

**Comment by gsoltis** [comment]
Hmm, I was not able to reproduce this. To make sure I understand, you're running some pipeline that has tasks that depend on `//:outdated`, `//:outdated` is failing, but the tasks that depend on it are continuing to run?

Note that there may be tasks that run *in parallel* with `//:outdated` that continue to produce output before `turbo` shuts them down.

**Comment by AddictArts** [comment]
> Hmm, I was not able to reproduce this. To make sure I understand, you're running some pipeline that has tasks that depend on `//:outdated`, `//:outdated` is failing, but the tasks that depend on it are continuing to run?
> 
> Note that there may be tasks that run _in parallel_ with `//:outdated` that continue to produce output before `turbo` shuts them down.

Correct all the dependsOn tasks keep going and don't stop.

**Comment by gsoltis** [comment]
In that example, `"dependsOn": ["//#outdated", "lint", "test", "dev"],`, an error in `lint, test` or `//#outdated` *shouldn't* prevent `dev` from _starting_. As written, `dev` doesn't depend on `lint, test` or `//#outdated`.

What should happen, if, e.g. `lint`, encounters an error: the `lint` task exits, `turbo` will not start any new tasks, and tasks that are already running will be stopped. Note that there is a race condition here, some tasks may complete before they can be stopped.

The final result from `turbo` should show `run failed` with some additional details and have a non-zero exit code. Are you seeing that output? Or are you seeing success reported?

Can you provide a repro similar to your setup? For a failing task, something like `"scripts": { "build": "exit 2" },` should be fine.

**Comment by AddictArts** [comment]
Thanks @gsoltis but `dev` does depend on `lint, and test`

```json
    "dev": {
      "dependsOn": ["lint", "test"],
```

**Comment by gsoltis** [comment]
@AddictArts can you provide a sample repository demonstrating the issue, along with the `turbo` command you're running?

**Comment by nathanhammond** [comment]
@AddictArts starting from here would be fantastic:

```
npx create-turbo@latest
cd my-turborepo
```

Then create a commit that creates the repository state you're looking for, and then a commit that introduces a failing script.

Our test process will be:
- Clone the repository.
- Check out HEAD~1.
- Run the build.
- Check out HEAD.
- Run the build.

**Comment by killpowa** [comment]
I also have a very similar issue in my repo. I have a build workspace that depends on the build of many other workspaces, yet if any of those other workspaces fail to build the overall turborepo process will succeed instead of failing.

**Comment by nathanhammond** [comment]
@killpowa can you get us a reproduction? We've attempted to investigate this and have not been able to reproduce the failure as described.

**Comment by anthonyshew** [comment]
Closing due to inactivity and stale version. If you're still running into this problem, please file a new issue.
